tools/cmake/utility.cmake

```diff
@@ -0,0 +1,68 @@
+function(set_macro_definitions)
+    set(multiValueArgs TARGETS REQUIRED OPTIONAL FILES_TO_CHECK)
+    cmake_parse_arguments(MACRO_DEFINITIONS "" "" "${multiValueArgs}" ${ARGN})
+    # Check for missing parameters.
+    if(NOT DEFINED MACRO_DEFINITIONS_TARGETS)
+        message("At least one target is required when setting macro definitions through CMake.")
+        return()
+    endif()
+    # Default file to check is demo_config.h
+    if(NOT DEFINED MACRO_DEFINITIONS_FILES_TO_CHECK)
+        set(MACRO_DEFINITIONS_FILES_TO_CHECK "demo_config.h")
+    endif()
+    set(CMAKE_REQUIRED_DEFINITIONS "")
+    set(DEFINED_MACROS "")
+    set(MISSING_REQUIRED_MACROS "")
+    foreach(application_target ${MACRO_DEFINITIONS_TARGETS})
+        if(DEFINED MACRO_DEFINITIONS_OPTIONAL)
+            foreach(optional_macro_definition ${MACRO_DEFINITIONS_OPTIONAL})
+                if(DEFINED ${optional_macro_definition})
+                    # Compile the application with the macro definition if it is defined.
+                    target_compile_definitions(
+                        ${application_target} PRIVATE
+                            ${optional_macro_definition}="${${optional_macro_definition}}"
+                    )
+                    list(APPEND DEFINED_MACROS "${required_macro_definition}")
```

Comment:
Should the optional macro (`optional_macro_definition`) be used here instead?

Reply:
Thanks for catching this